### ISSUE TITLE
[BUG]: discord-connector crash

### ISSUE THREAD

**Issue by ReshiramZekrom1** [openmultiplayer/server-beta-old#8]
# Description

I can't run the server with the discord-connector plugin. Even tried to run it alone with a blank gamemode and crashdetect without any differences. On the SA-MP Server everything is running fine.

I'm still using server.cfg if could matter something.

# How to re-produce this bug

1. Add "discord-connector.so" in the plugins list 
2. Run omp-server

# Relevant log output

from the terminal
```
[20/04/2022 21:57:20] [Info] Loading plugin: crashdetect.so
[20/04/2022 21:57:20] [Info]   CrashDetect plugin 4.19.4
[20/04/2022 21:57:20] [Info] Loading plugin: discord-connector.so

[log-core] server crash detected: fatal signal '11' (SIGSEGV) caught

[20/04/2022 21:57:20] [Info] [debug] Server crashed due to an unknown error

```

log.txt
```
[20/04/2022 21:57:20] [Info] Loading plugin: crashdetect.so
[20/04/2022 21:57:20] [Info]   CrashDetect plugin 4.19.4
[20/04/2022 21:57:20] [Info] Loading plugin: discord-connector.so
[20/04/2022 21:57:20] [Info] [debug] Server crashed due to an unknown error
[20/04/2022 21:57:20] [Info] [debug] Native backtrace:
[20/04/2022 21:57:20] [Info] [debug] #0 f4d298b6 in _Z13GetStackTraceRSt6vectorI10StackFrameSaIS0_EEPv () in plugins/crashdetect.so
[20/04/2022 21:57:20] [Info] [debug] #1 f4d21120 in _ZN18CrashDetectHandler20PrintNativeBacktraceERSoRKN2os7ContextE () in plugins/crashdetect.so
[20/04/2022 21:57:20] [Info] [debug] #2 f4d21886 in _ZN18CrashDetectHandler20PrintNativeBacktraceERKN2os7ContextE () in plugins/crashdetect.so
[20/04/2022 21:57:20] [Info] [debug] #3 f4d21f96 in _ZN18CrashDetectHandler7OnCrashERKN2os7ContextE () in plugins/crashdetect.so
[20/04/2022 21:57:20] [Info] [debug] #4 f4d28f82 in ?? () in plugins/crashdetect.so
[20/04/2022 21:57:20] [Info] [debug] #5 f7736cb0 in __kernel_rt_sigreturn () in linux-gate.so.1
[20/04/2022 21:57:20] [Info] [debug] #6 f49ff18d in ?? () in plugins/discord-connector.so
[20/04/2022 21:57:20] [Info] [debug] #7 081e825f in BIO_ctrl () in ./omp-server
[20/04/2022 21:57:20] [Info] [debug] Registers:
[20/04/2022 21:57:20] [Info] [debug] EAX: 00000000 EBX: f4bed000 ECX: f49ff170 EDX: f2b01bd8
[20/04/2022 21:57:20] [Info] [debug] ESI: 00000000 EDI: 00000000 EBP: 0000008a ESP: f34f9d50
[20/04/2022 21:57:20] [Info] [debug] EIP: f49ff18d EFLAGS: 00010286
[20/04/2022 21:57:20] [Info] [debug] Stack:
[20/04/2022 21:57:20] [Info] [debug] ESP+00000000: f49ff179 084d4000 f2b01bd8 081e825f
[20/04/2022 21:57:20] [Info] [debug] ESP+00000020: f34f9ed8 f774d5f0 00000000 f2b01bd8
[20/04/2022 21:57:20] [Info] [debug] ESP+00000040: f2b01bd8 0000008a 00000000 f2b01c50
[20/04/2022 21:57:20] [Info] [debug] ESP+00000060: f34f9dfc 00000000 f2b0048c 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000080: 00000000 00000000 00000000 f2b00470
[20/04/2022 21:57:20] [Info] [debug] ESP+000000a0: 00000000 00000000 f2b006b8 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+000000c0: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+000000e0: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000100: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000120: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000140: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000160: 00000000 00000000 00000000 ef20a700
[20/04/2022 21:57:20] [Info] [debug] ESP+00000180: 0abb5c88 084d4000 f34fa338 f487182a
[20/04/2022 21:57:20] [Info] [debug] ESP+000001a0: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+000001c0: 00000000 00000000 00000000 f34fa070
[20/04/2022 21:57:20] [Info] [debug] ESP+000001e0: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000200: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000220: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000240: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000260: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000280: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+000002a0: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+000002c0: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+000002e0: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000300: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000320: f34fa088 00000001 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000340: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000360: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+00000380: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+000003a0: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+000003c0: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] ESP+000003e0: 00000000 00000000 00000000 00000000
[20/04/2022 21:57:20] [Info] [debug] Loaded modules:
[20/04/2022 21:57:20] [Info] [debug] 00000000 - 004bc307 omp-server
[20/04/2022 21:57:20] [Info] [debug] f7736000 - f7736e2e linux-gate.so.1
[20/04/2022 21:57:20] [Info] [debug] f771f000 - f77219c4 /lib/i386-linux-gnu/libdl.so.2
[20/04/2022 21:57:20] [Info] [debug] f7702000 - f771d75b /lib/i386-linux-gnu/libpthread.so.0
[20/04/2022 21:57:20] [Info] [debug] f76f9000 - f770047c /lib/i386-linux-gnu/librt.so.1
[20/04/2022 21:57:20] [Info] [debug] f76a4000 - f76f8120 /lib/i386-linux-gnu/libm.so.6
[20/04/2022 21:57:20] [Info] [debug] f7686000 - f76a2354 /lib/i386-linux-gnu/libgcc_s.so.1
[20/04/2022 21:57:20] [Info] [debug] f74ce000 - f768c73b /lib/i386-linux-gnu/libc.so.6
[20/04/2022 21:57:20] [Info] [debug] f7738000 - f775c06c /lib/ld-linux.so.2
[20/04/2022 21:57:20] [Info] [debug] f71cd000 - f7223b8c components/Classes.so
[20/04/2022 21:57:20] [Info] [debug] f6f34000 - f72037f1 components/LegacyNetwork.so
[20/04/2022 21:57:20] [Info] [debug] f6edb000 - f6f38f04 components/TextLabels.so
[20/04/2022 21:57:20] [Info] [debug] f6e28000 - f6e9a57c components/Objects.so
[20/04/2022 21:57:20] [Info] [debug] f6cc8000 - f6e35105 components/LegacyConfig.so
[20/04/2022 21:57:20] [Info] [debug] f6c3f000 - f6cd0da4 components/Unicode.so
[20/04/2022 21:57:20] [Info] [debug] f6ab7000 - f6c3abbf /usr/lib/i386-linux-gnu/libstdc++.so.6
[20/04/2022 21:57:20] [Info] [debug] f6a67000 - f6abab98 components/Variables.so
[20/04/2022 21:57:20] [Info] [debug] f6a0c000 - f6a6bd2c components/TextDraws.so
[20/04/2022 21:57:20] [Info] [debug] f6903000 - f6957240 components/Checkpoints.so
[20/04/2022 21:57:20] [Info] [debug] f68ad000 - f6907f44 components/Actors.so
[20/04/2022 21:57:20] [Info] [debug] f6811000 - f685fe0c components/Timers.so
[20/04/2022 21:57:20] [Info] [debug] f67c1000 - f6815084 components/Dialogs.so
[20/04/2022 21:57:20] [Info] [debug] f676b000 - f67c55f0 components/Menus.so
[20/04/2022 21:57:20] [Info] [debug] f65f5000 - f677b538 components/Databases.so
[20/04/2022 21:57:20] [Info] [debug] f62c3000 - f6670f15 components/Pawn.so
[20/04/2022 21:57:20] [Info] [debug] f625b000 - f62cb904 components/Vehicles.so
[20/04/2022 21:57:20] [Info] [debug] f5fd7000 - f615a3ad components/Console.so
[20/04/2022 21:57:20] [Info] [debug] f5f83000 - f5fdb4ac components/GangZones.so
[20/04/2022 21:57:20] [Info] [debug] f56e4000 - f573c750 components/Pickups.so
[20/04/2022 21:57:20] [Info] [debug] f4d0a000 - f4d4e3ac /[...]/ius_omp/plugins/crashdetect.so
[20/04/2022 21:57:20] [Info] [debug] f4793000 - f4c228a6 /[...]/plugins/discord-connector.so
[20/04/2022 21:57:20] [Info] [debug] f457e000 - f479e348 /[...]/plugins/../log-core2.so
```

# open.mp server version

public beta build 1

# Operating system or distribution

Debian 9 VPS, kernel version is: 4.9.0-16-amd64

# Contact information

ReshiramZekrom#7809

# Additional information

///

**Comment by DEV-Catborisovv** [comment]
this is beta, the developer wrote that plugins with dynamic memory change may not work

**Comment by Hual** [comment]
> this is beta, the developer wrote that plugins with dynamic memory change may not work

discord-connector isn't a memory hacking plugin; it's most likely to do with openssl

**Comment by DEV-Catborisovv** [comment]
> > this is beta, the developer wrote that plugins with dynamic memory change may not work
> 
> discord-connector isn't a memory hacking plugin; it's most likely to do with openssl

ok, sorry, could you tell me if there will be a client release in the next few months?

**Comment by ReshiramZekrom1** [comment]
I managed to get it work. I had to build both the discord-connector and the log-core2 plugins with my VPS. I lost a couple of hours to understand how to actually build it since the instructions in the wiki are useless but luckily I've found out that the process was explained in detail in [this issue](https://github.com/maddinat0r/samp-discord-connector/issues/104#issuecomment-541840089). I've also updated OpenSSL to the version 1.1.1g and installed the libboost library.

Here there are my compiled builds, in case anyone may need them in the future:
[discord-connector.so](https://github.com/ReshiramZekrom1/samp-discord-connector/releases/download/v0.3.6/discord-connector.so)
[log-core2.so](https://github.com/ReshiramZekrom1/samp-discord-connector/releases/download/v0.3.6/log-core2.so)

I guess I can close the issue since it seems I'm the only one who have faced it, although everything was fine with the samp-server package. With those builds it seems to work fine now.
